### PR TITLE
Add bitwiseAnd() ops

### PR DESCRIPTION
**Pull request by fengwuyao**
- Implemented the bitwiseAnd() ops 
- Added the kernel in CPU backend
- Example usage:
```
const x = tf.tensor1d([0, 5, 3, 14]);
const y = tf.tensor1d([5, 0, 7, 11]);
tf.bitwiseAnd(x, y).print();

### CODE REVIEW

**Review by mattsoulanille — CHANGES_REQUESTED** [review]

File: tfjs-backend-cpu/src/kernels/BitwiseAnd.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+export const bitwiseAndImpl =
+  createSimpleBinaryKernelImpl(((a: number, b: number) => {
+    let res = 0;
+    for (let i = 0; i < 32; i++) {
+      if ((a & (1 << i)) && (b & (1 << i))) {
+        res |= (1 << i);
+      }
+    }
+    return res;
```
[the remark this answers is not in the document]
Comment (fengwuyao) [reply tfjs-backend-cpu/src/kernels/BitwiseAnd.ts]: Yeah, it should be just `a & b`. Thanks for pointing this out.

File: .vscode/settings.json [hunk 2]
```diff
@@ -18,7 +18,8 @@
   "editor.tabSize": 2,
   "editor.insertSpaces": true,
   "[typescript]": {
-    "editor.formatOnSave": true
+    "editor.formatOnSave": true,
+    "editor.defaultFormatter": "xaver.clang-format"
```
Comment (mattsoulanille) [inline .vscode/settings.json:22]: Thanks for fixing this!

**Review by mattsoulanille — APPROVED** [review]
LGTM!

File: tfjs-backend-cpu/src/shared.ts [hunk 3]
```diff
@@ -19,6 +19,7 @@
 export {simpleAbsImpl} from './kernels/Abs';
 export {addImpl} from './kernels/Add';
 export {bincountImpl, bincountReduceImpl} from './kernels/Bincount_impl';
+export {bitwiseAndImpl} from './kernels/BitwiseAnd';
```
Comment (Linchenn) [inline tfjs-backend-cpu/src/shared.ts:22]: Nice catch. Adding it here means other backend could use this kernel implementation. We could discuss this later when implementing this kernel for other backends.

File: tfjs-core/src/ops/bitwise_and.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+function bitwiseAnd_<R extends Rank>(
+    x: Tensor|TensorLike, y: Tensor|TensorLike): Tensor<R> {
+  const $x = convertToTensor(x, 'x', 'bitwiseAnd');
+  const $y = convertToTensor(y, 'y', 'bitwiseAnd');
+
+  if (!arraysEqual($x.shape, $y.shape)) {
+    throw new Error(`BitwiseAnd: Tensors must have the same shape. x: ${
+        $x.shape}, y: ${$y.shape}`);
+  }
+  if ($x.dtype !== 'float32' || $y.dtype !== 'float32') {
```
Comment (Linchenn) [inline tfjs-core/src/ops/bitwise_and.ts:57]: should it be 'int32'?

File: tfjs-core/src/ops/bitwise_and.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    throw new Error(
+        `BitwiseAnd: Only supports 'float32' values in tensor, found type of x: ${
```
Comment (Linchenn) [inline tfjs-core/src/ops/bitwise_and.ts:59]: int32

File: tfjs-core/src/ops/bitwise_and.ts (hunk 4 above)
Comment (fengwuyao) [reply tfjs-core/src/ops/bitwise_and.ts]: Thanks for pointing out. Updated.

File: tfjs-core/src/ops/bitwise_and.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+import {op} from './operation';
+
+/**
+ * Bitwise `AND` operation for input tensors.
+ *
+ * Given two input tensors, returns a new tensor
+ * with the `AND` calculated values.
+ *
+ * The method supports float32 values
```
Comment (Linchenn) [inline tfjs-core/src/ops/bitwise_and.ts:34]: nit

**Review by Linchenn — APPROVED** [review]
Just a nit. LGTM, this is very fast! Thanks!

**Pull request merged**